pkg/scheduler/internal/cache/cache.go

```diff
@@ -280,9 +280,11 @@ func (cache *schedulerCache) updateNodeInfoSnapshotList(snapshot *Snapshot, upda
 	if updateAll {
 		// Take a snapshot of the nodes order in the tree
 		snapshot.nodeInfoList = make([]*framework.NodeInfo, 0, cache.nodeTree.numNodes)
-		cache.nodeTree.resetExhausted()
-		for i := 0; i < cache.nodeTree.numNodes; i++ {
-			nodeName := cache.nodeTree.next()
+		nodesList, err := cache.nodeTree.list()
+		if err != nil {
+			klog.Error(err.Error())
+		}
+		for _, nodeName := range nodesList {
```

Comment:
whether the loop against `nodesList` here is needed when you get a error from `cache.nodeTree.list()`?

Reply:
The idea was that an error in listing the node would not fully prevent the scheduler to work properly, it would work with whatever partial list of nodes it got, and log an error to not let it go unnoticed. Is that problematic ?

Reply:
sound good, thank for the response.